src/zenml/core/base_component.py

```diff
@@ -66,15 +69,37 @@ def __init__(self, **values: Any):
 
         # Initialize values from the above sources.
         super().__init__(**values)
+        self._save_backup_file_if_required()
+
+    def _save_backup_file_if_required(self) -> None:
+        """Saves a backup of the config file if the schema changed."""
+        if self._superfluous_options:
+            logger.warning(
+                "Found superfluous configuration values for class `%s`: %s",
+                self.__class__.__name__,
+                set(self._superfluous_options),
+            )
+            config_path = self.get_serialization_full_path()
+            if os.path.exists(config_path):
+                backup_path = config_path + ".backup"
+                shutil.copyfile(config_path, backup_path)
```

Comment:
Does it make sense to just use path_utils here

Reply:
Yep good idea